### PR TITLE
Add Minutely probe for Unicorn

### PR DESCRIPTION
**Pull request by jvanbaarsen**
This adds a probe that will send the amount of active workers as a
custom metric.

The dashboard that can be used for this would look like this:

```yaml
-
  title: Unicorn
  graphs:
    -
      title: 'Worker count'
      line_label: Workers
      kind: gauge
      format: number
      fields:
        - unicorn_workers
```

### CODE REVIEW

**Comment by tombruijn** [comment]
You'll need to rebase on develop to fix those JRuby builds.

File: lib/appsignal/hooks/unicorn.rb [hunk 1]
```diff
@@ -38,5 +39,13 @@ def close
         end
       end
     end
+
+    class UnicornProbe
+      def call
+        ObjectSpace.each_object(Unicorn::HttpServer) do |s|
+          Appsignal.set_gauge("unicorn_workers", s.worker_processes)
```
Comment (tombruijn) [inline lib/appsignal/hooks/unicorn.rb]:
```suggestion
          Appsignal.set_gauge("unicorn_worker_count", s.worker_processes)
```

What naming do we want to follow here? At least singular names as discussed before. So do we add `_count` to these kinds of metrics?

File: lib/appsignal/hooks/unicorn.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def dependencies_present?
       end
 
       def install
+        Appsignal::Minutely.probes.register :unicorn_probe, UnicornProbe
```
Comment (tombruijn) [inline lib/appsignal/hooks/unicorn.rb]: I don't think we need to suffix this with `_probe`
```suggestion
        Appsignal::Minutely.probes.register :unicorn, UnicornProbe
```

File: lib/appsignal/hooks/unicorn.rb (hunk 1 above)
Comment (matsimitsu) [reply lib/appsignal/hooks/unicorn.rb]: I think so, we use `queue_length`, `worker_count` should be similar

**Comment by tombruijn** [comment]
@jvanbaarsen can you rebase on develop. That way the build is accurate again

**Comment by jvanbaarsen** [comment]
@tombruijn Done

**Comment by tombruijn** [comment]
@jvanbaarsen looks like JRuby doesn't work with ObjectSpace by default?

See the build on Travis: https://travis-ci.org/appsignal/appsignal-ruby/jobs/499838152

```
RuntimeError:
       ObjectSpace is disabled; each_object will only work with Class, pass -X+O to enable
```

**Comment by jvanbaarsen** [comment]
@tombruijn Interesting. That would require some diving into. I can do that later this week.

**Comment by tombruijn** [comment]
Moving it back to WIP for now.

**Comment by tombruijn** [comment]
Closing. Supporting JRuby is a hassle and it's just one metric.

**Pull request closed without merging**